x-pack/plugins/apm/public/hooks/useUrlParams.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useReducer } from 'react';
+import { useLocation } from './useLocation';
+import {
+  urlParamsReducer,
+  TimeRange,
+  refreshTimeRange,
+  resolveUrlParams,
+  IUrlParams
+} from '../store/urlParams';
+
+type RefreshTimeRangeFunction = (time: TimeRange) => void;
+
+export function useUrlParams(): [IUrlParams, RefreshTimeRangeFunction] {
+  const location = useLocation();
+  const [params, dispatch] = useReducer(
+    urlParamsReducer,
+    resolveUrlParams(location)
+  );
+
+  function refresh(time: TimeRange) {
+    dispatch(refreshTimeRange(time));
+  }
+
+  return [params, refresh];
```

Comment:
I think it should return an object: `return { urlParams, refresh };`
It makes sense for generic hooks like `useState` to return an array so the consumer can name the values however they want. However, in this case `useUrlParams` is not generic. It takes no arguments and always returns the same values, and I think it could create more confusion if we mix how we use it:

```js
const [params, dispatch] = useUrlParams() 
...
const [urlParams, refresh] = useUrlParams() 
...
const [value, fn] = useUrlParams() 
```

Reply:
That's fair, I went back and forth on that.